homeassistant/components/generic/config_flow.py

```diff
@@ -183,14 +183,31 @@ async def async_test_still(
     except (
         TimeoutError,
         RequestError,
-        HTTPStatusError,
         TimeoutException,
     ) as err:
         _LOGGER.error("Error getting camera image from %s: %s", url, type(err).__name__)
         return {CONF_STILL_IMAGE_URL: "unable_still_load"}, None
+    except HTTPStatusError as err:
+        _LOGGER.error(
+            "Error getting camera image from %s: %s %s",
+            url,
+            type(err).__name__,
+            err.response.text,
+        )
+        if err.response.status_code == 401:
+            return {CONF_STILL_IMAGE_URL: "unable_still_load_401"}, None
+        if err.response.status_code == 403:
+            return {CONF_STILL_IMAGE_URL: "unable_still_load_403"}, None
+        if err.response.status_code == 404:
+            return {CONF_STILL_IMAGE_URL: "unable_still_load_404"}, None
+        if err.response.status_code == 500:
+            return {CONF_STILL_IMAGE_URL: "unable_still_load_500"}, None
+        if err.response.status_code == 503:
+            return {CONF_STILL_IMAGE_URL: "unable_still_load_503"}, None
+        return {CONF_STILL_IMAGE_URL: "unable_still_load"}, None
```

Comment:
In the default state we don't return the status code.  Should we append it anyway so at least the user gets the code?  If so this section could be simplified to return

f"unable_still_load_{err.response.status_code}"

Reply:
I don't think that would work, the string here is actually just a key to define which string from strings.json is shown to the user. 

So a parameter in here would mean we'd need a lot of strings.json entries.

The error code will now be shown in the logs.

Reply:
Got, did not understand those were translation strings.